djongo/models/fields.py

```diff
@@ -70,177 +67,248 @@ def __getattr__(self, name):
 
     @property
     def _client(self):
-        return (
-            pymongo_connections[self.db]
+        return (pymongo_connections[self.db]
                 .cursor()
-                .db_conn[self.model._meta.db_table]
-        )
+                .db_conn[self.model._meta.db_table])
 
 
 class FormlessField(Field):
-    empty_strings_allowed = False
-
-    def formfield(self, **kwargs):
-        raise TypeError(
-            'A Formless Field cannot be modified from Django Admin.'
-        )
-
-
-class ListField(FormlessField):
     """
-    MongoDB allows the saving of python lists as BSON Array type data. The `ListField` is useful in such cases.
+    Allows for the inclusion of an instance of an abstract model as
+    a field inside a document.
     """
     empty_strings_allowed = False
+    base_type = dict
 
-    def get_db_prep_value(self, value, connection, prepared=False):
-        if not isinstance(value, list):
-            raise ValueError(
-                f'Value: {value} must be of type list'
-            )
-        return value
-
-    def to_python(self, value):
-        if not isinstance(value, list):
-            raise ValueError(
-                f'Value: {value} stored in DB must be of type list'
-                'Did you miss any Migrations?'
-            )
-        return value
-
-
-class DictField(FormlessField):
-    """
-    MongoDB allows the saving of python dicts as BSON object type data. The `DictField` is useful in such cases.
-    """
-    empty_strings_allowed = False
+    def __init__(self,
+                 model_container: typing.Type[Model],
+                 *args, **kwargs):
+        self.model_container = model_container
+        self._validate_container()
+        super().__init__(*args, **kwargs)
 
-    def get_db_prep_value(self, value, connection, prepared=False):
-        if not isinstance(value, dict):
-            raise ValueError(
-                f'Value: {value} must be of type dict'
-            )
-        return value
+    def _validate_container(self):
+        for field in self.model_container._meta.get_fields():
```

Comment:
@nesdis Will this cause problems while starting the app? Feels like this may throw a `Models aren't loaded yet.` on starting the server because get_fields() requires all models to be ready.